From 06e81737440edd96eaf5f3399fafa9baa25197f3 Mon Sep 17 00:00:00 2001
From: Matti Kosola <matti.kosola@jolla.com>
Date: Tue, 16 Dec 2014 14:41:15 +0200
Subject: [PATCH] Use mmap for fastboot data

Signed-off-by: Matti Kosola <matti.kosola@jolla.com>
---
 fastboot/fastboot.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/fastboot/fastboot.c b/fastboot/fastboot.c
index d8a878b..e512108 100644
--- a/fastboot/fastboot.c
+++ b/fastboot/fastboot.c
@@ -39,6 +39,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/mman.h>
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/types.h>
@@ -173,11 +174,12 @@ static void *load_fd(int fd, unsigned *_sz)
         goto oops;
     }
 
-    data = (char*) malloc(sz);
+    //data = (char*) malloc(sz);
+    data = mmap(NULL, sz, PROT_READ, MAP_SHARED, fd, 0);
     if(data == 0) goto oops;
 
-    if(read(fd, data, sz) != sz) goto oops;
-    close(fd);
+    //if(read(fd, data, sz) != sz) goto oops;
+    //close(fd);
 
     if(_sz) *_sz = sz;
     return data;
-- 
1.9.1

